src/main/webapp/dev/js/main/instructorHelp.js

```diff
@@ -25,7 +34,319 @@ function bindPanelCollapseLinksAndAnchor() {
     });
 }
 
+/**
+ * Constructs a hashmap of questions and utilizes it to
+ * build an index for search. Each question also has associated
+ * tags which are obtained from stemming and sub-categories allocated
+ * to different questions.
+ */
+function prepareQuestionsForSearch() {
+    let questionCount = 0;
+
+    function addQuestionsForTopic(topic) {
+        const topicContentElement = $(topic).next();
+
+        topicContentElement.find('.panel-group').each((idx, subTopic) => {
+            const subTopicText = $(subTopic).prev().text();
+            const subTopicQuestions = $(subTopic).children();
+
+            let subTopicTags = subTopicText.split(' ').map((word) => {
+                // ensure case insensitivity
+                const lowerCaseWord = word.toLowerCase();
+
+                // ignore 1 and 2 letter words which are probably articles and prepositions
+                if (lowerCaseWord.length < 3) {
+                    return '';
+                }
+                return elasticlunr.stemmer(lowerCaseWord);
+            });
+            subTopicTags = $.grep(subTopicTags, (tag) => {
+                if (tag !== '') {
+                    return true;
+                }
+                return false;
+            });
+
+            for (let i = 0; i < subTopicQuestions.length; i += 1) {
+                const question = $(subTopicQuestions[i]);
+                const htmlId = question.attr('id');
+                const tagsSeenSoFar = {};
+
+                let questionTags = subTopicTags.concat(htmlId.split('-'));
```

Comment:
Merge two arrays?
`jQuery.merge()`?

Why the tags should care about the htmlID?

Reply:
question ids provide relevant keywords regarding the question which can added as tags eg. `student-view-responses`

Reply:
Why not extract keywords from question title rather than question id?

Reply:
we'll need to process and tokenize it since it contains lots of irrelevant words and prepositions. This info is readily available from the ids so i thought just use it as is

Reply:
Okay.